packages/provider/src/lib/types/Payload.ts

```diff
@@ -157,5 +152,507 @@ export namespace Payload {
+     * @since 1.0.0
+     */
+    export interface ByValue<StoredValue> extends Payload, ByValueWithPath<Primitive>, Payload.Data<Data<StoredValue>> {
+      method: Method.Partition;
+    }
+
+    /**
+     * The data for {@link Method.Partition}
+     * @since 1.0.0
+     */
+    export interface Data<StoredValue> {
+      /**
+       * The truthy entries.
+       * @since 1.0.0
+       */
+      truthy: Record<string, StoredValue>;
+
+      /**
+       * The falsy entries.
+       * @since 1.0.0
+       */
+      falsy: Record<string, StoredValue>;
+    }
+  }
+
+  /**
+   * The payload for {@link Method.Push}
+   * @since 1.0.0
+   */
+  export interface Push<Value> extends Payload, KeyPath {
+    method: Method.Push;
+
+    /**
+     * The value to push to an array.
+     * @since 1.0.0
+     */
+    value: Value;
+  }
+
+  /**
+   * The payload for {@link Method.Random}
+   * @since 1.0.0
+   */
+  export interface Random<StoredValue> extends Payload, Data<StoredValue[]> {
+    method: Method.Random;
+
+    /**
+     * The amount of random values to get.
+     * @since 1.0.0
+     */
+    count: number;
+
+    /**
+     * Whether to allow duplicate values.
+     * @since 1.0.0
+     */
+    duplicates: boolean;
+  }
+
+  /**
+   * The payload for {@link Method.RandomKey}
+   * @since 1.0.0
+   */
+  export interface RandomKey extends Payload, Data<string[]> {
+    method: Method.RandomKey;
+
+    /**
+     * The amount of random keys to get.
+     * @since 1.0.0
+     */
+    count: number;
+
+    /**
+     * Whether to allow duplicate keys.
+     * @since 1.0.0
+     */
+    duplicates: boolean;
+  }
+
+  /**
+   * The payload for {@link Method.Remove}
+   * @since 1.0.0
+
+
+   */
+  export type Remove<Value> = Remove.ByHook<Value> | Remove.ByValue;
+
+  export namespace Remove {
+    /**
+     * The hook payload for {@link Method.Remove}
+     * @since 1.0.0
+     */
+    export interface ByHook<Value> extends Payload, KeyPath, Payload.ByHook<Hook<Value, boolean>> {
+      method: Method.Remove;
+
+      type: Type.Hook;
+    }
+
+    /**
+     * The value payload for {@link Method.Remove}
+     * @since 1.0.0
+     */
+    export interface ByValue extends Payload, KeyPath, Payload.ByValue<Primitive> {
+      method: Method.Remove;
+
+      type: Type.Value;
+    }
+  }
+
+  /**
+   * The payload for {@link Method.Set}
+   * @since 1.0.0
+   */
+  export interface Set<Value> extends Payload, KeyPath {
+    method: Method.Set;
+
+    /**
+     * The value to set.
+     * @since 1.0.0
+     */
+    value: Value;
+  }
+
+  /**
+   * The payload for {@link Method.SetMany}
+   * @since 1.0.0
+   */
+  export interface SetMany extends Payload {
+    method: Method.SetMany;
+
+    /**
+     * Whether to overwrite existing entries.
+     * @since 1.0.0
+     */
+    overwrite: boolean;
+
+    /**
+     * The entries to set.
+     * @since 1.0.0
+     */
+    entries: SetMany.KeyPathValue[];
+  }
+
+  export namespace SetMany {
```

Comment:
Description needed?

Reply:
No, this is the namespace, see the interface.